src/main/java/frc/robot/commands/drive/Drive.java

```diff
@@ -15,24 +20,60 @@
      */
     private boolean sideFlipped = false;
 
+    private double pitchDegrees;
+
     public Drive() {
         requires(Drivetrain.getInstance());
     }
 
+    private void correctForPitch(double[] stickSpeeds) {//x-tip
+        //if its over 45 there's no point in correcting it
+        if (Math.abs(pitchDegrees) < 45) {
+            //correctionFactor keeps the tilt correction within a certain threshold so it doesn't correct too much
+            double correctionFactor = (RobotMap.DRIVE.MAX_CORRECTION_FACTOR - RobotMap.DRIVE.MIN_CORRECTION_FACTOR) / (45 - RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES);
+
+            double correctionOffset = correctionFactor * (pitchDegrees - RobotMap.DRIVE.PITCH_THRESHOLD_DEGREES);
+            stickSpeeds[0] += correctionOffset;
+            stickSpeeds[1] += correctionOffset;
+            RobotMath.normalize(stickSpeeds);
+        }
+    }
+
+    private void updatePitchStatus() {
+        this.pitchDegrees = NavX.getInstance().getPitchDegrees();
```

Comment:
It seems saner to just do this assignment in `execute` directly (less jumping around both in reading and, as a lesser problem, in actual execution)